core/services/relay/evm/evmtesting/chain_reader_interface_tester.go

```diff
@@ -0,0 +1,327 @@
+						},
+					},
+					MethodReturningUint64: {
+						ChainSpecificName: "getPrimitiveValue",
+					},
+					DifferentMethodReturningUint64: {
+						ChainSpecificName: "getDifferentPrimitiveValue",
+					},
+					MethodReturningUint64Slice: {
+						ChainSpecificName: "getSliceValue",
+					},
+					EventName: {
+						ChainSpecificName: "Triggered",
+						ReadType:          types.Event,
+						OutputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{Fields: map[string]string{"NestedStruct.Inner.IntVal": "I"}},
+						},
+					},
+					EventWithFilterName: {
+						ChainSpecificName: "Triggered",
+						ReadType:          types.Event,
+						EventInputFields:  []string{"Field"},
+					},
+					triggerWithDynamicTopic: {
+						ChainSpecificName: triggerWithDynamicTopic,
+						ReadType:          types.Event,
+						EventInputFields:  []string{"fieldHash"},
+						InputModifications: codec.ModifiersConfig{
+							&codec.RenameModifierConfig{Fields: map[string]string{"FieldHash": "Field"}},
+						},
+					},
+					triggerWithAllTopics: {
+						ChainSpecificName: triggerWithAllTopics,
+						ReadType:          types.Event,
+						EventInputFields:  []string{"Field1", "Field2", "Field3"},
+					},
+					MethodReturningSeenStruct: {
+						ChainSpecificName: "returnSeen",
+						InputModifications: codec.ModifiersConfig{
+							&codec.HardCodeModifierConfig{
+								OnChainValues: map[string]any{
+									"BigField": testStruct.BigField.String(),
+									"Account":  hexutil.Encode(testStruct.Account),
+								},
+							},
+							&codec.RenameModifierConfig{Fields: map[string]string{"NestedStruct.Inner.IntVal": "I"}},
+						},
+						OutputModifications: codec.ModifiersConfig{
+							&codec.HardCodeModifierConfig{OffChainValues: map[string]any{"ExtraField": AnyExtraValue}},
+							&codec.RenameModifierConfig{Fields: map[string]string{"NestedStruct.Inner.IntVal": "I"}},
+						},
+					},
+				},
+			},
+			AnySecondContractName: {
+				ContractABI: chain_reader_tester.ChainReaderTesterMetaData.ABI,
+				Configs: map[string]*types.ChainReaderDefinition{
+					MethodReturningUint64: {
+						ChainSpecificName: "getDifferentPrimitiveValue",
+					},
+				},
+			},
+		},
+	}
+	it.client = it.Helper.Client(t)
+
+	it.deployNewContracts(t)
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) Name() string {
+	return "EVM"
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) GetAccountBytes(i int) []byte {
+	account := [20]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+	account[i%20] += byte(i)
+	account[(i+3)%20] += byte(i + 3)
+	return account[:]
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) GetChainReader(t T) clcommontypes.ChainReader {
+	ctx := it.Helper.Context(t)
+	if it.cr != nil {
+		return it.cr
+	}
+
+	lggr := logger.NullLogger
+	db := it.Helper.NewSqlxDB(t)
+	lpOpts := logpoller.Opts{
+		PollPeriod:               time.Millisecond,
+		FinalityDepth:            4,
```

Comment:
I left this the same as it was before I did refactoring figuring that I could make tests run in a reasonable time.  Would it make more sense for me to make this configurable so that things like finality breaches could be caught by the test?  Does anyone know the impact of waiting for finality?  For example, how long would it take to deploy a contract then send four transactions and wait for their events on Sepolia?

Reply:
If we are only testing the ChainReader interface, I don't think finality matters so much as returning valid structured data. While the ChainReader interface tests are sort of integration tests with the underlying log poller, adjusting finality here seems to just duplicate logpoller tests. Maybe I'm missing something?

Reply:
Agreed, that's why I left it, I just wanted to present the the option and where it could benefit us.  I'm not sure there would be much gain anyways, because finality breaches aren't constant anyway.  So the odds you get one during a test seem close to zero.